app/Jobs/CreateQueryserviceBatchesJob.php

```diff
@@ -0,0 +1,97 @@
+{
+    private const NAMESPACE_ITEM = 120;
+    private const NAMESPACE_PROPERTY = 122;
+    private const NAMESPACE_LEXEME = 146;
+
+    private int $entityLimit;
+
+    public function __construct()
+    {
+        $this->entityLimit = Config::get('wbstack.qs_batch_entity_limit');
+    }
+
+    public function handle(): void
+    {
+        DB::transaction(function () {
+            $lastCheckpoint = QsCheckpoint::get();
+
+            // Get events after the point that batch was created
+            // TODO maybe filter by NS here?
+            $events = EventPageUpdate::where(
+                'id', '>', $lastCheckpoint
+            )->get();
+
+            $wikiBatchesEntities = [];
+            $lastEventId = 0;
+            foreach ($events as $event) {
+                if (
+                    $event->namespace == self::NAMESPACE_ITEM ||
+                    $event->namespace == self::NAMESPACE_PROPERTY ||
+                    $event->namespace == self::NAMESPACE_LEXEME
+                ) {
+                    $wikiBatchesEntities[$event->wiki_id][] = $event->title;
+                }
+                if ($event->id > $lastEventId) {
+                    $lastEventId = $event->id;
+                }
+            }
+
+            $notDoneBatches = QsBatch::where([
+                ['done', '=', 0],
+                ['pending_since', '=', null],
+                ['failed', '=', false],
+            ])->get();
+
+            // Insert the newly created batches into the table...
+            foreach ($wikiBatchesEntities as $wikiId => $entityIdsFromEvents) {
+                // If we already have a not done batch for this same wiki, then merge that into a new batch
+                foreach ($notDoneBatches as $qsBatch) {
+                    if ($qsBatch->wiki_id !== $wikiId) {
+                        continue;
+                    }
+
+                    $entitiesOnBatch = explode(',', $qsBatch->entityIds);
+                    $tentativeMerge = array_unique(array_merge($entityIdsFromEvents, $entitiesOnBatch));
+                    if (count($tentativeMerge) > $this->entityLimit) {
+                        continue;
+                    }
+
+                    $qsBatch->update([
+                        'entityIds' => implode(',', $tentativeMerge),
+                        'eventFrom' => $lastCheckpoint,
+                        'eventTo'=> $lastEventId,
+                    ]);
+                    // after updating, we need to skip the creation below
+                    // so we continue the outer loop instead
+                    continue 2;
+                }
+
+                $chunks = array_chunk($entityIdsFromEvents, $this->entityLimit);
+                foreach ($chunks as $chunk) {
+                    // Insert the new batch
+                    QsBatch::create([
+                        'done' => 0,
+                        'eventFrom' => $lastCheckpoint,
+                        'eventTo'=> $lastEventId,
```

Comment:
From what I understand we could now drop these columns. Not sure if we should as they could be helpful when debugging?